src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/CancelModal/CancelModal.tsx

```diff
@@ -1,69 +1,164 @@
 
   const payload: CancelExpenditurePayload = {
     colonyAddress: colony.colonyAddress,
     expenditure,
     userAddress: user?.walletAddress ?? '',
   };
+  const motionPayload: ExpenditureCancelMotionPayload = {
+    colony,
+    expenditure,
+    userAddress: user?.walletAddress ?? '',
+    motionDomainId: Id.RootDomain,
+    votingReputationAddress: votingReputationAddress ?? '',
+  };
+
+  const {
+    stakeAmount = '0',
+    isLoading,
+    stakedExpenditureAddress = '',
+  } = useExpenditureStaking();
 
   const cancelExpenditure = useAsyncFunction({
     submit: ActionTypes.EXPENDITURE_CANCEL,
     error: ActionTypes.EXPENDITURE_CANCEL_ERROR,
     success: ActionTypes.EXPENDITURE_CANCEL_SUCCESS,
   });
+  const cancelExpenditureViaMotion = useAsyncFunction({
+    submit: ActionTypes.MOTION_EXPENDITURE_CANCEL,
+    error: ActionTypes.MOTION_EXPENDITURE_CANCEL_ERROR,
+    success: ActionTypes.MOTION_EXPENDITURE_CANCEL_SUCCESS,
+  });
+  const cancelStakedExpenditure = useAsyncFunction({
+    submit: ActionTypes.STAKED_EXPENDITURE_CANCEL,
+    error: ActionTypes.STAKED_EXPENDITURE_CANCEL_ERROR,
+    success: ActionTypes.STAKED_EXPENDITURE_CANCEL_SUCCESS,
+  });
+  const cancelStakedExpenditureViaMotion = useAsyncFunction({
+    submit: ActionTypes.MOTION_STAKED_EXPENDITURE_CANCEL,
+    error: ActionTypes.MOTION_STAKED_EXPENDITURE_CANCEL_ERROR,
+    success: ActionTypes.MOTION_STAKED_EXPENDITURE_CANCEL_SUCCESS,
+  });
+  const reclaimExpenditureStake = useAsyncFunction({
+    submit: ActionTypes.RECLAIM_EXPENDITURE_STAKE,
+    error: ActionTypes.RECLAIM_EXPENDITURE_STAKE_ERROR,
+    success: ActionTypes.RECLAIM_EXPENDITURE_STAKE_SUCCESS,
+  });
 
-  const handleFundExpenditure = async () => {
+  const handleFundExpenditure = async ({ decisionMethod, penalise }) => {
     setIsSubmitting(true);
     try {
       if (!expenditure) {
         return;
       }
 
-      await cancelExpenditure(payload);
+      const stakedPayload: CancelStakedExpenditurePayload = {
+        colonyAddress: colony.colonyAddress,
+        expenditure,
+        stakedExpenditureAddress,
+        shouldPunish: penalise === PenaliseOptions.Yes,
+      };
+      const stakedMotionPayload: StakedExpenditureCancelMotionPayload = {
+        colonyAddress: colony.colonyAddress,
+        colonyName: colony.name,
+        expenditure,
+        motionDomainId: Id.RootDomain,
+        stakedExpenditureAddress,
+        shouldPunish: penalise === PenaliseOptions.Yes,
+      };
+      const reclaimPayload: ReclaimExpenditureStakePayload = {
+        colonyAddress: colony.colonyAddress,
+        nativeExpenditureId: expenditure.nativeId,
+      };
+
+      if (
+        decisionMethod &&
+        decisionMethod.value === DecisionMethod.Reputation
+      ) {
+        if (penalise) {
+          await cancelStakedExpenditureViaMotion(stakedMotionPayload);
+        } else {
+          await cancelExpenditureViaMotion(motionPayload);
+        }
```

Comment:
Which saga to use should not depend on `penalise` but rather on the expenditure type - if staked, `cancelStakedExpenditureViaMotion` should always be used. You can pass the `penalise` as `shouldPunish` param to the saga.